packages/cli/src/templates/expo.ts

```diff
@@ -0,0 +1,255 @@
+import chalk from "chalk";
+import { Listr } from "listr2";
+import { existsSync } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+import { TemplateOptions } from "../commands/create";
+import { Template } from "./template";
+
+import { exec } from "node:child_process";
+import { promisify } from "node:util";
+const execAsync = promisify(exec);
+export interface ExpoContext {
+  options: TemplateOptions;
+}
+
+export const Expo: Template = {
+  name: "expo",
+  description: "A new expo typescript project with gluestack-ui renderer",
+  async handler(options) {
+    await new Listr<ExpoContext>([
+      {
+        title: "Create directory",
+        task: async ({ options: { cwd } }, task) => {
+          task.title += ` ${cwd}`;
+          if (existsSync(cwd)) {
+            throw new Error(`Directory ${cwd} already exists`);
+          }
+          await execAsync(
+            `npx -y create-expo-app ${cwd} -t expo-template-blank-typescript`,
+          );
+        },
+      },
+      {
+        title: "Create Expo project",
+        task: async ({ options: { cwd } }) => {
+          try {
+            const appJsonFilePath = `${cwd}/app.json`;
+            const data = await readFile(appJsonFilePath, "utf8");
+            const config = JSON.parse(data);
+            config.expo.jsEngine = "jsc";
+            const updatedConfig = JSON.stringify(config, undefined, 2);
+            await writeFile(appJsonFilePath, updatedConfig, "utf8");
+          } catch (error) {
+            console.error("Error reading or writing the app.json file:", error);
+          }
+
+          try {
+            const babelConfigFilePath = `${cwd}/babel.config.js`;
+            const data = await readFile(babelConfigFilePath, "utf8");
+            const updatedConfig = data.replace(
+              `presets: ['babel-preset-expo'],`,
+              `presets: ['babel-preset-expo'],
+              plugins: [
+                ["@babel/plugin-transform-class-properties", { "loose": true }],
+                ["@babel/plugin-transform-private-methods", { "loose": true }],
+                ["@babel/plugin-transform-private-property-in-object", { "loose": true }]
```

Comment:
My own testing showed that only the `@babel/plugin-transform-private-methods` felt necessary:

```suggestion
                ["@babel/plugin-transform-private-methods", { "loose": true }]
```